### ISSUE TITLE
Can't open a directory named "checkpoints"

### ISSUE THREAD

**Issue by yangql176** [jupyterlab/jupyterlab#12408]
## Description

Can't open a directory named "checkpoints"
<img width="341" alt="image" src="https://user-images.githubusercontent.com/50950740/163379195-9e0997e7-4f0f-4bca-96bb-78cfc6f524ed.png">

## Reproduce

1. Create a new directory named **"checkpoints"**
2. Open it
3. Nothing happened

## Expected behavior

Open this directory correctly.

## Context

- Operating System and version: 
- Browser and version: 
- JupyterLab version:  3.1.12
## Cause of this bug

These two methods are the reason why this happened.
![image](https://user-images.githubusercontent.com/50950740/163377727-22df8702-76d6-4a53-81e1-10a5d8daf427.png)

The function **listCheckpoints** is triggered when a directory named checkpoints is opened.
![image](https://user-images.githubusercontent.com/50950740/163378328-3b75233c-ef7f-402c-9a76-8b29cadca315.png)

The _checkpoints_ parameter should be with a special character instead of a potential directory name .

<details><summary>Troubleshoot Output</summary>
<pre>
Paste the output from running `jupyter troubleshoot` from the command line here.
You may want to sanitize the paths in the output.
</pre>
</details>

<details><summary>Command Line Output</summary>
<pre>
Paste the output from your command line running `jupyter lab` here, use `--debug` if possible.
</pre>
</details>

<details><summary>Browser Output</summary>

<pre>
Paste the output from your browser Javascript console here, if applicable.

</pre>
</details>

**Comment by jweill-aws** [comment]
Thank you for your report! If we can't fix this, the limitation on directories named `checkpoints` needs to be documented.

**Comment by yangql176** [comment]
> Thank you for your report! If we can't fix this, the limitation on directories named `checkpoints` needs to be documented.

Sure. Thanks!

**Comment by jtpio** [comment]
This also sounds related to https://github.com/jupyter/notebook/issues/5569

**Comment by yangql176** [comment]
Hello @jweill-aws 😊.  May I ask how's the progress on this "checkpoint" issue🤔️

**Comment by jweill-aws** [comment]
@yangql176 I don't believe that anyone is actively working on this issue. This issue is accepted, which means that any developer can submit a pull request to fix it.

**Comment by yangql176** [comment]
> @yangql176 I don't believe that anyone is actively working on this issue. This issue is accepted, which means that any developer can submit a pull request to fix it.

Hi, me again. Since this problem does affect the normal folders opening, we think it should be repaired as soon as possible. My back-end colleagues and I think it maybe a reasonable way to modify the URL of the entire checkpoints related functions, such as changing it into a URL request that is independent of the file path. **However, this needs to be modified by both the front end and the back end. It needs to be fixed in this repo and the[ server-end repo](https://github.com/jupyter-server/jupyter_server) at the same time**. 

 I don't know you guys' plan. What do you think?

**Comment by yangql176** [comment]
@jweill-aws  BTW,  we can't open the dir named "trust" either，which should also pay attention to.

**Comment by jweill-aws** [comment]
@yangql176 This is an open source project being worked on by people at various organizations around the world. I do not know if anyone is actively working on this bug. If this bug is affecting your ability to use JupyterLab and Jupyter Server, I invite you to start a discussion or a pull request to fix it. The [Jupyter Contributing documentation](https://docs.jupyter.org/en/latest/contributing/content-contributor.html) and the [Jupyter Server contributor guide](https://jupyter-server.readthedocs.io/en/latest/contributors/contributing.html) may help you out here.

**Comment by l365692902** [comment]
I'm temporarily using symbolic links to work around this issue, like `ln -s checkpoints checkpoints_link`, hoping it can help some people.
